go-controller/pkg/node/gateway_localnet.go

```diff
@@ -215,28 +194,13 @@ func (l *localPortWatcher) addService(svc *kapi.Service) error {
 				if utilnet.IsIPv6String(externalIP) != isIPv6Service {
 					continue
 				}
-				if _, exists := l.localAddrSet[externalIP]; exists {
-					iptRules = append(iptRules, getExternalIPTRules(port, externalIP, ip)...)
-					klog.V(5).Infof("Will add iptables rule for ExternalIP: %s", externalIP)
-				} else if l.networkHasAddress(net.ParseIP(externalIP)) {
-					klog.V(5).Infof("ExternalIP: %s is reachable through one of the interfaces on this node, will skip setup", externalIP)
-				} else {
-					if gatewayIP != "" {
-						routeUsage[externalIP] = true
-					} else {
-						klog.Warningf("No gateway of appropriate IP family for ExternalIP %s for Service %s/%s",
-							externalIP, svc.Namespace, svc.Name)
-					}
-				}
-			}
-		}
-		for externalIP := range routeUsage {
-			if stdout, stderr, err := util.RunIP("route", "replace", externalIP, "via", gatewayIP, "dev", types.K8sMgmtIntfName, "table", localnetGatewayExternalIDTable); err != nil {
-				klog.Errorf("Error adding routing table entry for ExternalIP %s, via gw: %s: stdout: %s, stderr: %s, err: %v", externalIP, gatewayIP, stdout, stderr, err)
-			} else {
-				klog.Infof("Successfully added route for ExternalIP: %s", externalIP)
+
+				iptRules = append(iptRules, getExternalIPTRules(port, externalIP, ip)...)
+				klog.V(5).Infof("Will add iptables rule for ExternalIP: %s", externalIP)
```

Comment:
we should have a better log here that includes the service name like "Adding iptables rules for service: %s with external IP: %s"

Reply:
fixed